### PR TITLE
Don't run synchronization code on a player with autoplay blocked and add an autoplay warning

### PR DESCRIPTION
**Pull request by lfzawacki**

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/external-video-player/component.jsx [hunk 1]
```diff
@@ -21,14 +29,27 @@ class VideoPlayer extends Component {
...
     this.opts = {
       controls: isPresenter,
+      file: {
+        attributes: {
+          controls: isPresenter,
+        },
+      },
+      dailymotion: {
+        params: {
+          controls: isPresenter,
```
[the remark this answers is not in the document]
Comment (lfzawacki) [reply bigbluebutton-html5/imports/ui/components/external-video-player/component.jsx]: The idea was to hide the seek controls from the viewer, but I actually forgot about the volume for dailymotion. I'll fix it.

File: bigbluebutton-html5/imports/ui/components/external-video-player/container.jsx [hunk 2]
```diff
@@ -1,28 +1,18 @@
...
-  },
-});
-
 const ExternalVideoContainer = props => (
   <ExternalVideo {...{ ...props }} />
 );
 
-export default injectIntl(withTracker(({ intl, isPresenter }) => {
-  const title = intl.formatMessage(intlMessages.title);
+export default withTracker(({ intl, isPresenter }) => {
```
[the remark this answers is not in the document]
Comment (lfzawacki) [reply bigbluebutton-html5/imports/ui/components/external-video-player/container.jsx]: :+1:

**Comment by lfzawacki** [comment]
> I was testing with the different sources and when I initially shared a YouTube video, the presenter client showed the "blocked by autoplay" message, but the media was playing behind the message fine. I tried to reproduce, but I wasn't successful so I'm unsure how it got triggered. When I stopped and started the playback again the message cleared up.
> 
> I also had an issue when I was trying to play a Twitch stream. I'm not sure if it was autoplay or not, but the stream was just black (no preview) and I had to pause the stream and restart to get it playing. The viewer clients showed the preview fine.

Hey Chad, can you try to reproduce the errors you saw on top of these changes? I think there could still be problems if the presenter or viewer is trying to watch a high quality video and the connection is not able to keep up and download it fast enough, but I'm not sure how to fix it or if we can do anything at all.

File: bigbluebutton-html5/imports/ui/components/external-video-player/component.jsx [hunk 3]
```diff
@@ -151,8 +198,16 @@ class VideoPlayer extends Component {
...
+        logger.debug({ logCode: 'external_video_presenter_ready' }, 'Presenter is ready to sync');
+
+        if (!this.state.hasPlayedBefore) {
+          this.setState({playing: true});
+        }
+      });
+
       onMessage('playerUpdate', (data) => {
-        if (!this.player) {
+        if (!this.player || !this.state.hasPlayedBefore) {
```
Comment (capilkey) [inline bigbluebutton-html5/imports/ui/components/external-video-player/component.jsx]: I just noticed all of the `this.state.X` usage. They should be using destructuring.
```
const {
  X,
} = this.state;
```

Comment (lfzawacki) [reply bigbluebutton-html5/imports/ui/components/external-video-player/component.jsx]: Addressed this in the newest commits

**Comment by lfzawacki** [comment]
> Another issue that I noticed is that if the video is playing, and the presenter reloads with autoplay blocked, the viewers will still be playing, but the video will be constantly set back to 0.
> 
> There's also an issue with autoplay being blocked on mobile where the label blocks the play button.
> image

Both issues addressed in the newest commits, the warning is a little smaller in mobile screen sizes and you can click through it.

**Pull request merged**